src/internal/http/listener/httpslistener_test.go

```diff
@@ -0,0 +1,229 @@
+package httpslistener_test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/Shopify/sarama"
+	"github.com/dfds/confluent-gateway/configuration"
+	httpslistener "github.com/dfds/confluent-gateway/internal/http/listener"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	topic1 = "i.exist.and.people.listen.to.me"
+	topic2 = "i.also.exist.and.a.thing.listens"
+
+	consumer1 = "consumer1-prod"
+	consumer2 = "consumer2-prod"
+	consumer3 = "consumer3-prod"
+	consumer4 = "consumer4-dev"
+)
+
+type channelBuffer struct {
+	cin  chan bool
+	cout chan bool
+}
+
+func (cb *channelBuffer) Run() {
+	go func() {
+		<-cb.cin
+		cb.cout <- true
+	}()
+}
+
+type mockConsumerGroupHandler struct{}
+
+func (m *mockConsumerGroupHandler) Setup(session sarama.ConsumerGroupSession) error { return nil }
+func (m *mockConsumerGroupHandler) Cleanup(sarama.ConsumerGroupSession) error       { return nil }
+func (m *mockConsumerGroupHandler) ConsumeClaim(session sarama.ConsumerGroupSession, claim sarama.ConsumerGroupClaim) error {
+	for message := range claim.Messages() {
+		session.MarkMessage(message, "")
+		//session.Commit()
+	}
+	return nil
+}
+
+type responseData struct {
+	Topic     string   `json:"topic"`
+	Consumers []string `json:"consumers"`
+}
+
+func must[R interface{}](result R, err error) R {
+	if err != nil {
+		panic(err)
+	}
+	return result
+}
+
+func runConsumer(ctx context.Context, group sarama.ConsumerGroup, topic string) {
+	go func() {
+		if err := group.Consume(ctx, []string{topic}, &mockConsumerGroupHandler{}); err != nil {
+			panic(err)
+		}
+	}()
+}
+
+func TestConsumerCount(t *testing.T) {
+	/*
+	** SETUP
+	 */
+	config := configuration.LoadInto("../../../", &configuration.Configuration{})
+
+	kafkaConfig := sarama.NewConfig()
+	kafkaConfig.Producer.Return.Successes = true
+	kafkaConfig.Consumer.Offsets.Initial = sarama.OffsetOldest
+	prodclient := must(sarama.NewClient([]string{config.KafkaClusterProdBroker}, kafkaConfig))
+	prodadmin := must(sarama.NewClusterAdminFromClient(prodclient))
+	defer prodadmin.Close()
+	devclient := must(sarama.NewClient([]string{config.KafkaClusterDevBroker}, kafkaConfig))
+	devadmin := must(sarama.NewClusterAdminFromClient(devclient))
```

Comment:
Just a suggestion: 
use camelCase for these variable names
```
prodClient
prodAdmin
devClient
devAdmin
```

Reply:
for sure; I prefer that myself and this is a leftover of my wanting to test things quickly